src/main/java/filter/lexer/Lexer.java

```diff
@@ -21,6 +21,7 @@
             // These have higher priority than Symbol
             new Rule("\\d{4}-\\d{1,2}-\\d{1,2}", TokenType.DATE), // YYYY-MM?-DD?
             new Rule("[A-Za-z]+(-[A-Za-z]+)*\\s*:", TokenType.QUALIFIER),
+            new Rule("[A-Za-z0-9#][/A-Za-z0-9.'+-]+#", TokenType.COMPOUND_ID),
```

Comment:
I'm not sure about this, because it doesn't check that a number actually follows. This may lead to the error appearing at the `id` qualifier that is inserted when parsing, instead of at the `id` that contains the whole compound id.

Could we use something like `[a-z]+/[a-z]+#[0-9]+` (simple example) to tokenise it instead, then extract the id as well in `parseCompoundId`? A simple `value.split("/|#")` would work to get them out, since we've confirmed that it matches the pattern.